lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -2786,3 +2786,128 @@ def test_group_access(self, role_name, course_is_cohorted, thread_group_state):
             self.assertFalse(expected_error)
         except Http404:
             self.assertTrue(expected_error)
+
+
+@ddt.ddt
+class RetrieveThreadTest(
```

Comment:
For possible examples, you can see the `test_group_access` methods in this file.